share/rocm/cmake/ROCMConfig.cmake

```diff
@@ -2,4 +2,11 @@
 # Copyright (C) 2023 Advanced Micro Devices, Inc.
 # ######################################################################################################################
 
-find_package(ROCmCMakeBuildTools HINTS "${CMAKE_CURRENT_LIST_DIR}")
+get_filename_component(_new_rocmcmakebuildtools_path "${CMAKE_CURRENT_LIST_DIR}" DIRECTORY)
+get_filename_component(_new_rocmcmakebuildtools_path "${_new_rocmcmakebuildtools_path}" DIRECTORY)
+# two directories up is sufficient for windows search, but linux search requires the share directory
+get_filename_component(_new_rocmcmakebuildtools_path_linux "${_new_rocmcmakebuildtools_path}" DIRECTORY)
+
+find_package(ROCmCMakeBuildTools REQUIRED HINTS
+    "${_new_rocmcmakebuildtools_path}"
+    "${_new_rocmcmakebuildtools_path_linux}")
```

Comment:
Should this be `find_dependency`?

Reply:
@cgmb I've updated it to pass through the `REQUIRED` and `QUIET` arguments if provided, and the version argument if provided as well. Can you take another look?

Reply:
As an aside, the `REQUIRED` flag was used originally as `ROCmCMakeBuildTools` should always be installed alongside this file, and so the `REQUIRED` was effectively a short-cut for erroring out if it couldn't be found (indicating that something is wrong).

Reply:
I looked at your comment prior to your edit and saw that you probably intended to suggest `find_dependency`. I have updated to use this, as it obviates the previous work I did to forward the arguments.

Reply:
Good for catching that. Sorry for being unclear. I tried to add the HINTS with an edit and it seems I accidentally overwrote the core point via a copy/paste error.